source/portable/NetworkInterface/NXP1060/NetworkInterface.c

```diff
@@ -0,0 +1,797 @@
+                }
+                else
+                {
+                    break;
+                }
+            }
+            else
+            {
+                eEMACState = xEMAC_Ready;
+            }
+
+        /* Fall through. */
+        case xEMAC_Ready:
+            FreeRTOS_printf( ( "Driver ready for use." ) );
+
+            /* Kick the task once the driver is ready. */
+            if( receiveTaskHandle != NULL )
+            {
+                xTaskNotify( receiveTaskHandle, DRIVER_READY, eSetValueWithOverwrite );
+            }
+
+            xResult = pdPASS;
+
+            break;
+    }
+
+    return xResult;
+}
+
+BaseType_t xNetworkInterfaceOutput( NetworkBufferDescriptor_t * const pxNetworkBuffer,
+                                    BaseType_t xReleaseAfterSend )
+{
+    status_t result;
+    BaseType_t xReturn = pdFAIL;
+
+    if( bGlobalLinkStatus == true )
+    {
+        /* ENET_SendFrame copies the data before sending it. Therefore, the network buffer can
+         * be released without worrying about the buffer memory being used by the ENET_SendFrame
+         * function. */
+        result = ENET_SendFrame( ethernetifLocal->base,
+                                 &ethernetifLocal->handle,
+                                 pxNetworkBuffer->pucEthernetBuffer,
+                                 pxNetworkBuffer->xDataLength,
+                                 0,
+                                 false,
+                                 NULL );
+
+        switch( result )
+        {
+            case kStatus_ENET_TxFrameBusy:
+                FreeRTOS_printf( ( "Failed to send the frame - driver busy!\r\n" ) );
+                break;
+
+            case kStatus_Success:
+                iptraceNETWORK_INTERFACE_TRANSMIT();
+                xReturn = pdPASS;
+                break;
+        }
+    }
+
+    if( xReleaseAfterSend == pdTRUE )
+    {
+        vReleaseNetworkBufferAndDescriptor( pxNetworkBuffer );
+    }
+
+    return xReturn;
+}
+
+static void prvEMACHandlerTask( void * parameter )
+{
+    bool LinkUp = false;
+    status_t readStatus;
+
+    /* Wait for the driver to finish starting. */
+    ( void ) ulTaskNotifyTake( pdTRUE, portMAX_DELAY );
+
+    while( pdTRUE )
+    {
+        if( ulTaskNotifyTake( pdTRUE, pdMS_TO_TICKS( 500 ) ) == pdFALSE )
+        {
+            /* No RX packets for a bit so check for a link. */
+            const IPStackEvent_t xNetworkEventDown = { .eEventType = eNetworkDownEvent, .pvData = NULL };
+
+            do
+            {
+                readStatus = PHY_GetLinkStatus( &phyHandle, &LinkUp );
+
+                if( readStatus == kStatus_Success )
+                {
+                    if( LinkUp == pdFALSE )
+                    {
+                        /* The link is down. */
+                        bGlobalLinkStatus = false;
+                        /* We need to setup the PHY again. */
+                        eEMACState = xEMAC_SetupPHY;
```

Comment:
There's no need to reconfigure the phy.

```suggestion
                        eEMACState = xEMAC_WaitPHY;
```